### PR TITLE
Allow a list of values for multi-option question predicates

### PR DESCRIPTION
**Pull request by natsid**
### Description
Now predicates based on multi-options questions can actually have lists as the `values` field on the right hand side! Works for both single select (radios, dropdowns) and multi select questions.

* Use a set of checkbox inputs for the `value` field when the left hand side of the predicate operator is a selection or selections from a multi-option question.
* Parse these checkbox values differently than if we were getting one string value.
* Removed  "is equal to" and "is not equal to" operator options from single-select predicates. This is because then we'd need to limit the number of checkboxes selected in that case. We could have added extra validation to enforce selecting only a single option in that case. But, since "is one of" with just one value is equivalent to "is equal to" and likewise for "is none of" and "is not equal to", we think this strategy actually _helps_ the user by not getting in their way. If they select multiple options on the right side, they probably mean "is one of" anyway.  A validation error when a user uses "is equal to" and multiple options on the right hand side seems not very helpful.
* Note that `string` scalars can already have "lists" on the right-hand side prior to this PR, but the value is just entered as a comma-separated list in that case.

Screen Shot 2021-06-24 at 2 01 35 PM

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Manually tested this flow with both single select and multi select question predicates
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Works toward #322

### CODE REVIEW

File: universal-application-tool-0.0.1/app/assets/javascripts/main.ts [hunk 1]
```diff
@@ -165,31 +165,55 @@ function configurePredicateForm(event: Event) {
...
 
+function shouldHideOperator(
+  selectedScalarType: string,
+  selectedScalarValue: string,
+  operatorOption: HTMLOptionElement
+): boolean {
+  // If this operator is not for the currently selected type, hide it.
+  return (
+    !(selectedScalarType in operatorOption.dataset) ||
+    // Special case for SELECTION scalars (which are of type STRING):
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/assets/javascripts/main.ts:204]: praise: thank you for the detailed docstring!

File: universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java [hunk 2]
```diff
@@ -163,13 +178,11 @@ private PredicateValue parsePredicateValue(Scalar scalar, Operator operator, Str
             return PredicateValue.of(Long.parseLong(value));
         }
 
-      default: // STRING, LIST_OF_STRINGS
+      default: // STRING, should not include LIST_OF_STRINGS since that should be covered at the top
+        // of the method with `values`.
         switch (operator) {
           case ANY_OF:
           case NONE_OF:
-          case IN:
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java:170]: question: I'm not sure I understand why these operators disappear. I thought all of these have a list of strings on the right?

File: universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java [hunk 3]
```diff
@@ -127,6 +128,69 @@ public void update_withValidFormData_savesNewPredicate() {
         .doesNotContain("This block is always shown.");
   }
 
+  @Test
+  public void update_withSingleSelectQuestion() {
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java:132]: praise: thanks for adding tests!

File: universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java [hunk 4]
```diff
@@ -127,6 +128,69 @@ public void update_withValidFormData_savesNewPredicate() {
...
+            .build();
+
+    Result result = controller.update(request, programWithThreeBlocks.id, 3L);
+
+    assertThat(result.status()).isEqualTo(SEE_OTHER);
+    assertThat(result.redirectLocation())
+        .hasValue(
+            routes.AdminProgramBlockPredicatesController.edit(programWithThreeBlocks.id, 3L).url());
+    assertThat(result.flash().get("error")).isEmpty();
+    assertThat(result.flash().get("success").get()).contains("Saved visibility condition blah");
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java:156]: question: blah?

**Review by carolinedanzi — APPROVED** [review]
Thanks for doing this! I think it makes sense - so many edge cases!

File: universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java (hunk 4 above)
Comment (natsid) [reply universal-application-tool-0.0.1/test/controllers/admin/AdminProgramBlockPredicatesControllerTest.java:156]: LOL oh no. Thanks for the catch

File: universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java (hunk 2 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java:170]: They do all have list of strings on the right, that is correct! Let me see if I can explain this effectively in writing.

Once we get to this point, we know the scalar on the left is of type STRING, and that it is *not* of value SELECTION. That's because I added that `if` statement at the top that parses the `values` list rather than the `value` string. I updated the logic in that if statement so it's hopefully a little more clear what's going on.

Additionally, any `STRING` scalar type only has `ANY_OF`, `NONE_ON`, `EQUAL_TO`, `NOT_EQUAL_TO` as operator options. The other ones (`IN`, etc) are not available unless there is a `LIST_OF_STRINGS` on the left hand side.

The outer switch statement is based on the scalar on the *left hand side*. I know this is kind of collaged together and awkward, but hopefully it works. :)
